### PR TITLE
Use !! notation instead of ValidateArg.NotNull

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: src/Microsoft.TestPlatform.ObjectModel/Utilities/AssemblyHelper.cs [hunk 1]
```diff
@@ -32,14 +32,11 @@ public static class AssemblyHelper
...
     /// </summary>
-    public static bool? DoesReferencesAssembly(string source, AssemblyName referenceAssembly)
+    public static bool? DoesReferencesAssembly(string source, AssemblyName referenceAssembly!!)
     {
         try
         {
             ValidateArg.NotNullOrEmpty(source, nameof(source));
-            ValidateArg.NotNull(referenceAssembly, nameof(referenceAssembly));
-
-            Debug.Assert(!string.IsNullOrEmpty(source));
```
Comment (Evangelink) [inline src/Microsoft.TestPlatform.ObjectModel/Utilities/AssemblyHelper.cs:42]: Unneeded assert as we throw if null or empty a few lines before.

Comment (Evangelink) [inline src/Microsoft.TestPlatform.ObjectModel/Utilities/AssemblyHelper.cs:42]: Unneeded assert as we throw if null or empty a few lines before.

**Review by nohwnd — APPROVED** [review]

**Pull request merged**